app/services/push_notification/identifier_recycled_event.rb

```diff
@@ -0,0 +1,25 @@
+module PushNotification
+  class IdentifierRecycledEvent
+    EVENT_TYPE = 'https://schemas.openid.net/secevent/risc/event-type/identifier-recycled'.freeze
+
+    attr_reader :user, :email
+
+    def initialize(user:, email:)
+      @user = user
+      @email = email
+    end
+
+    def event_type
+      EVENT_TYPE
+    end
+
+    def payload
+      {
+        subject: {
+          subject_type: 'email',
+          email: email,
+        },
+      }
+    end
```

Comment:
I'm curious others' thoughts on having separate `event_type` and `payload` methods. It would simplify the expected event API to have one payload, but it would be a super extra nested hash? It would look like this but IDK if others think this is weird:

```ruby
def events
  {
    EVENT_TYPE => {
      subject: {
        subject_type: 'email',
        email: email,
      }
    }
  }
end
```

Reply:
I think that the way you have it makes sense. The HttpPush object is responsible for figuring out how to handle transforming event types and payloads into the structure of the spec.

If in the future we need to send multiple events at once, I'd expect it to be the place where it can call `event.event_type` to do checks like seeing if there are duplicates of the same type, etc.

Reply:
Thanks! That makes sense too